src/darwin/Framework/CHIPTests/MTRFabricInfoTests.m

```diff
@@ -0,0 +1,283 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#import <Matter/Matter.h>
+
+// system dependencies
+#import <XCTest/XCTest.h>
+
+#import "MTRTestKeys.h"
+#import "MTRTestStorage.h"
+
+static uint16_t kTestVendorId1 = 0xFFF1u;
+static uint16_t kTestVendorId2 = 0xFFF2u;
+
+// Macro so that we get useful line numbers
+#define CHECK_FABRIC_INFO(fabricInfoList, expected)                                                                                \
```

Comment:
This should work just fine as a helper method instead of a macro? You'll still get the test that's failing and the location of the XCTAssert that's failing

Reply:
You would, but you would not know where in the test it is.  If there are multiple CheckFabricInfo checks in the test, you would not know which one failed.

Reply:
Found a way to log stacks; will update accordingly.